core/http_component.cxx

```diff
@@ -182,20 +258,91 @@ class http_component_impl
       session = std::move(s);
     }
 
-    auto op = std::make_shared<pending_http_operation>(io_, request);
-    op->start(
-      [callback = std::move(callback)](auto resp, auto ec) mutable {
-        callback(std::move(resp), ec);
-      },
-      [session_manager, session, service = request.service]() mutable {
-        session_manager->check_in(service, session);
-      });
-    op->send_to(session);
+    op->start([callback = std::move(callback)](auto resp, auto ec) mutable {
+      callback(std::move(resp), ec);
+    });
+    op->set_stream_end_callback([session_manager, session, service = request.service]() mutable {
+      session_manager->check_in(service, session);
+    });
 
+    if (!session->is_connected()) {
+      session_manager->connect_then_send_pending_op(
+        session,
+        {},
+#ifdef COUCHBASE_CXX_CLIENT_COLUMNAR
+        op->dispatch_deadline_expiry(),
+#endif
+        op->deadline_expiry(),
+        [op = std::move(op)](std::error_code ec, std::shared_ptr<io::http_session> http_session) {
```

Comment:
`op` shouldn't be moved here, since we are later returning it from `do_http_request` - we can just copy it instead

Reply:
Ahh, thanks!  Will update 👍